### DISCUSSION TITLE
Gallery Card

### DISCUSSION THREAD

**Discussion by mishoboss** [home-assistant/frontend#7603] [Dashboard]
Now once we have Media browser page in Lovelace, showing videos and pictures, it would be awesome if we have a gallery Lovelace card too. Something similar to this project: https://github.com/TarheelGrad1998/GalleryCard

Personally I would love to have a list of the last 10 CCTV motion detections next to each camera stream in the UI. And when clicked on a list item, to playback that motion detection record.

**Comment by KTibow** [comment]
Cards usually aren't that large. Are you suggesting a scroll layout that's small, or a whole new panel for cameras?

**Reply by mishoboss** [reply]
Well, personally I imagine that card rendered as a scrollable list. Each item in the list corresponds to a file in a given directory (Folder sensor). When clicked on an item in the list, a video player popup is displayed, like it is in Media browser.

**Reply by KTibow** [reply]
Would you mind making a mockup in MS Paint or something similar?

**Reply by mishoboss** [reply]
Personally I imagine the list something like the shopping list: https://www.home-assistant.io/lovelace/shopping-list/
Just a simple list in a scrollable container, so if items exceed container's height, container can scroll. List items should be sorted by date created, DESC. Maybe it makes sense to introduce a property to limit the number of showing items, e.g. always show up to 10 items only.

![image](https://user-images.githubusercontent.com/1629929/98419294-6ca28580-208d-11eb-9944-189c9dd1486e.png)

On item click a popup with video player or picture is opened.

**Reply by mishoboss** [reply]
The card could have a `view` property taking either `list` or `previews` values. The `list` option renders the list above, while `previews` renders thumbnails of the images or videos in the directory. Something like this:

![image](https://user-images.githubusercontent.com/1629929/98420330-9eb4e700-208f-11eb-824c-a106b9f542fa.png)

**Comment by iantrich** [comment]
I could see this being a potential view option for the media browser, but not sure about a Lovelace card.

**Reply by mikepan-ovoenergy** [reply]
IMHO, a Lovelace card would be really powerful. Media browser is cool, but it's not context related to anything of your UI. For example I would like to see the last 5 entries in the house on my security page where alarm panel card is located, together with the CCTV camera of the front door. Contextually this information belongs here.

**Reply by KTibow** [reply]
You can add a link to the media browser with the markdown card. Do you think the ability to embed a panel in a tab would help out with this?
